radicle-cli/src/commands/add.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::Context as _;
+use radicle::{prelude::Id, storage::WriteStorage, Profile};
+use radicle_crypto::PublicKey;
+
+use crate::terminal as term;
+
+pub fn run<S>(profile: &Profile, storage: &S, id: Id, key: PublicKey) -> anyhow::Result<()>
+where
+    S: WriteStorage,
+{
+    let signer = term::signer(profile)?;
+    let me = signer.public_key();
+
+    let mut project = storage
+        .get(&profile.public_key, id)?
+        .context("No project with such ID exists")?;
+
+    let repo = storage.repository(id)?;
+
+    if !project.is_delegate(me) && key != *me {
+        return Err(anyhow::anyhow!("'{}' is not a current delegate of the project and does not have permission to add another person's key. However, they may add their own.", me));
+    }
```

Comment:
We probably don't want this right? This lets anyone add their own key, even if they aren't a delegate :)

Reply:
So, initially, I thought the same. But is there any harm in them adding themselves as a proposal? If the current delegates don't agree then the document shouldn't verify.

Reply:
Yes but here it won't be a proposal, it's going to write that update to your *actual* identity branch for the project. I think for proposals we have to handle them quite differently, and in coordination with `rad patch` :thinking:

Reply:
Oh right, you did mention that :monocle_face: Hmmmm does this CLI command make any sense then, in hindsight? I'm thinking it might be worth seeing what you have in the identity RIP before going any deeper here, wdyt?

Reply:
It does make sense, as I said for a `threshold = 1` situation, where any delegate can update the doc with this command.

Reply:
So it's completely limited to the case where `threshold = 1`. As soon as one bumps that the new document is verified but if they make any changes it'll say that the quorum is not reached, right? How useful is that if it only covers a single case? :thinking:

Reply:
well no, you can just keep the threshold at `1` for ever. You get the github admin model then, where everyone in the delegate set is an "admin".

Reply:
I'd argue this is the most common case actually that we're able to handle, ie. how most ACLs are setup on github right now.

Reply:
Eww :upside_down_face: